android/src/main/java/ee/forgr/capacitor_updater/DownloadService.java

```diff
@@ -54,84 +45,111 @@ protected void onHandleIntent(Intent intent) {
+        progressFile.delete();
+        progressFile.createNewFile();
+        tempFile.createNewFile();
+        downloadedBytes = 0;
+      }
+
+      if (downloadedBytes > 0) {
+        httpConn.setRequestProperty("Range", "bytes=" + downloadedBytes + "-");
+      }
+
+      int responseCode = httpConn.getResponseCode();
+
+      if (responseCode == HttpURLConnection.HTTP_OK || responseCode == HttpURLConnection.HTTP_PARTIAL) {
+        String contentType = httpConn.getContentType();
+        int contentLength = httpConn.getContentLength() + (int) downloadedBytes;
+
+        InputStream inputStream = httpConn.getInputStream();
+        FileOutputStream outputStream = new FileOutputStream(tempFile, downloadedBytes > 0);
+
+        // Writing initial progression into file
+        if (downloadedBytes == 0) {
+          try (BufferedWriter writer = new BufferedWriter(new FileWriter(progressFile))) {
+            writer.write(String.valueOf(downloadedBytes));
+          }
+        }
+
+        int bytesRead = -1;
+        byte[] buffer = new byte[4096];
+        int lastPercent = 0;
+        while ((bytesRead = inputStream.read(buffer)) != -1) {
+
+          outputStream.write(buffer, 0, bytesRead);
+          downloadedBytes += bytesRead;
+
+          // Updating the progress file
+          try (BufferedWriter writer = new BufferedWriter(new FileWriter(progressFile))) {
+            writer.write(String.valueOf(downloadedBytes));
+          }
+
+          // Saving progress (flushing every 100 Ko)
+          if (downloadedBytes % 102400 == 0) {
+            outputStream.flush();
+          }
+
+          // Computing percentage
+          int percent = calcTotalPercent(downloadedBytes, contentLength);
+          if (percent != lastPercent) {
+            notifyDownload(id, percent);
+            lastPercent = percent;
           }
-          publishResults(dest, id, version, checksum, sessionKey, "");
         }
+
+        outputStream.close();
+        inputStream.close();
+
+        // Rename the temp file with the final name (dest)
+        tempFile.renameTo(new File(documentsDir, dest));
+        progressFile.delete();
+        publishResults(dest, id, version, checksum, sessionKey, "");
+      } else {
+        progressFile.delete();
       }
-    } catch (OutOfMemoryError e) {
-      e.printStackTrace();
-      publishResults("", id, version, checksum, sessionKey, "low_mem_fail");
-    } catch (Exception e) {
+      httpConn.disconnect();
+    } catch (IOException e) {
       e.printStackTrace();
-      publishResults(
-        "",
-        id,
-        version,
-        checksum,
-        sessionKey,
-        e.getLocalizedMessage()
-      );
+
     }
   }
 
   private void notifyDownload(String id, int percent) {
     Intent intent = new Intent(PERCENTDOWNLOAD);
     intent.setPackage(getPackageName());
     intent.putExtra(ID, id);
-    intent.putExtra(PERCENT, percent);
+    intent.putExtra("percent", percent);
```

Comment:
Why stop using the static?

Reply:
Just a mistake, fixed it back in 169d51b